dex/msgjson/types.go

```diff
@@ -941,6 +978,92 @@ func (n *PenaltyNote) Serialize() []byte {
 	return append(b, []byte(p.Details)...)
 }
 
+// Client should send bond info when their bond tx is fully-confirmed. Server
+// should start waiting for required confs when it receives the 'postbond'
+// request if the txn is found. Client is responsible for submitting 'postbond'
+// for their bond txns when they reach required confs, or when the client
+// restarts and finds bonds recorded in their DB that is flagged as neither
+// accepted nor expired (also maybe if not listed in the 'connect' response).
+
+// PostBond should include the unsigned bond transaction for validation prior to
+// broadcasting the signed transaction so the client hasn't needlessly locked
+// funds if the transaction is rejected.
```

Comment:
> If the transaction is rejected then the funds wouldn't be locked

The comment means if there was no unsigned txn being checked first.  We don't want a published+signed txn rejected.

> Server should check so that the bond is not rejected by the server?

Server first validates the transaction structure with the unsigned bond txns, then client submits it again with the confidence that it's good.

Also as you noted elsewhere, this initial comms round can be used to agree on other bond parameters.